barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -246,7 +264,11 @@ template <typename Curve> class IPA {
 
             // Step 6.d
             // Receive the challenge from the verifier
-            const Fr round_challenge = transcript->get_challenge<Fr>("IPA:round_challenge_" + index);
+            const Fr round_challenge = transcript->template get_challenge<Fr>("IPA:round_challenge_" + index);
+
```

Comment:
it seemed everywhere else we use ASSERTs. I know they only trigger in debug mode, but that should be fine for now, right? Or is there any specific reason why we want to go with `throw_or_abort`

Reply:
also we don't document anywhere why these cannot be zero right?

Reply:
My understanding is ASSERTs should be used to detect cases of programmers' mistakes. And in this case you'd want to abort in production,too

Reply:
Added docs